tests/connect_redirect/connect_redirect_tests.cpp

```diff
@@ -127,16 +208,50 @@ _initialize_test_globals()
     REQUIRE((v6_addresses == 0 || v6_addresses == 3));
     IN6ADDR_SETLOOPBACK((PSOCKADDR_IN6)&_globals.addresses[socket_family_t::IPv6].loopback_address);
 
+    _globals.mode = _get_execution_mode(_execution_mode_string);
+    _globals.user_token = _logon_user(_user_name, _password);
     _globals.destination_port = _destination_port;
     _globals.proxy_port = _proxy_port;
     _globals_initialized = true;
 }
 
+static void
+_validate_audit_map_entry(_In_ const struct bpf_object* object)
+{
+    bpf_map* audit_map = bpf_object__find_map_by_name(object, "audit_map");
+    REQUIRE(audit_map != nullptr);
+
+    fd_t map_fd = bpf_map__fd(audit_map);
+
+    uint64_t process_id = get_current_pid_tgid();
+    sock_addr_audit_entry_t entry = {0};
+    int result = bpf_map_lookup_elem(map_fd, &process_id, &entry);
+    REQUIRE(result == 0);
+
+    REQUIRE(process_id == entry.process_id);
+
+    SECURITY_LOGON_SESSION_DATA* data = NULL;
+    result = LsaGetLogonSessionData((PLUID)&entry.logon_id, &data);
+    REQUIRE(result == ERROR_SUCCESS);
+
+    if (_globals.protocol == IPPROTO_TCP) {
+        if (_globals.mode == Admin) {
+            REQUIRE(entry.is_admin == 1);
+        } else {
+            REQUIRE(entry.is_admin == 0);
+        }
+    }
+
+    LsaFreeReturnBuffer(data);
+}
+
 static void
 _load_and_attach_ebpf_programs(_Outptr_ struct bpf_object** return_object)
 {
     struct bpf_object* object = bpf_object__open("cgroup_sock_addr2.o");
+    printf("errno = %d\n", errno);
```

Comment:
Why the printf?  Should this be
```
REQUIRE(errno == 0);
```
?

Reply:
Yes I had added this for debugging. removed it now.